src/header.rs

```diff
@@ -154,15 +154,18 @@ fn convert_tuple_to_record_field(tuple: &[Value]) -> io::Result<Field> {
 fn convert_value_to_field_shape(field: &Value) -> io::Result<Vec<u64>> {
     if let Value::List(ref lengths) = *field {
         lengths.iter().map(convert_value_to_positive_integer).collect()
+    } else if let Value::Tuple(ref lengths) = *field {
+        lengths.iter().map(convert_value_to_positive_integer).collect()
     } else {
         Err(invalid_data("shape must be list or tuple"))
     }
 }
 
 fn convert_value_to_positive_integer(number: &Value) -> io::Result<u64> {
-    if let Value::Integer(number) = *number {
-        if number > 0 {
-            Ok(number as u64)
+    if let Value::Integer(number) = number {
+        let parts = number.to_u64_digits().1;
+        if !parts.is_empty() && parts[0] > 0 {
```

Comment:
Does this comparison not trivially succeed?  I would expect we should be testing the `num_bigint::Sign` returned by `to_u64_digits`.

I believe we also should check that the number is not larger than a `u64`, by checking the length of the digits.